src/code_08005e54.c

```diff
@@ -1,31 +1,228 @@
+    }
+    oarg1 = arg6 + (arg1 - oarg1);
+    arg6 = oarg1;
+    oarg2 = arg7 + (arg2 - oarg2);
+    arg7 = oarg2;
+    if (func_08005e54(arg5, &arg6, &arg7, &arg3, &arg4) == 0) {
+        return;
+    }
+    arg1 += arg6 - oarg1;
+    arg2 += arg7 - oarg2;
 
-#include "asm/code_08005e54/asm_0800637c.s"
+    temp_r2 = &arg0->unk0[arg1] + arg0->unk4 * arg2;
+    temp_r1 = &arg5->unk0[arg6] + arg5->unk4 * arg7;
 
-#include "asm/code_08005e54/asm_08006448.s"
+    for (i = 0; i < arg4; i++) {
+        for (j = 0; j < (arg3 >> 3); temp_r2 += 8, temp_r1 += 8, j++) {
+            temp_r1[0] = temp_r2[0];
+            temp_r1[1] = temp_r2[1];
+            temp_r1[2] = temp_r2[2];
+            temp_r1[3] = temp_r2[3];
+            temp_r1[4] = temp_r2[4];
+            temp_r1[5] = temp_r2[5];
+            temp_r1[6] = temp_r2[6];
+            temp_r1[7] = temp_r2[7];
+        }
+        for (j = 0; j < (arg3 & 7); temp_r2++, temp_r1++, j++) {
+            temp_r1[0] = temp_r2[0];
+        }
+        temp_r2 += arg0->unk4 - arg3;
+        temp_r1 += arg5->unk4 - arg3;
+    }
+}
+
+struct struct_08005e54 *func_08006328(u16 arg0, s32 arg1, s32 arg2) {
+    struct struct_08005e54 *temp;
+
+    temp = (struct struct_08005e54 *)mem_heap_alloc_id(arg0, sizeof(struct struct_08005e54));
+    temp->unk0 = (u16 *)mem_heap_alloc_id(arg0, arg2 * arg1 * sizeof(u16));
+    temp->unk4 = arg1;
+    temp->unk6 = arg2;
+    return temp;
+}
+
+void func_08006364(struct struct_08005e54 *arg0) {
+    mem_heap_dealloc(arg0->unk0);
+    mem_heap_dealloc(arg0);
+}
+
+void func_0800637c(struct struct_08005e54 *arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, u16 arg5) {
+    u32 i;
+    u32 j;
+    u16 *temp_r6;
+    u16 *temp_r1;
+    u16 temp_r4 = arg5 << 0xc;
+    u16 temp_r5 = 0xfff;
+
+    temp_r6 = &arg0->unk0[arg1] + arg2 * arg0->unk4;
+    for (i = 0; i < arg4; i++) {
+        temp_r1 = temp_r6;
+        for (j = 0; j < (arg3 >> 3); temp_r1 += 8, j++) {
+            temp_r1[0] = (temp_r1[0] & temp_r5) | temp_r4;
+            temp_r1[1] = (temp_r1[1] & temp_r5) | temp_r4;
+            temp_r1[2] = (temp_r1[2] & temp_r5) | temp_r4;
+            temp_r1[3] = (temp_r1[3] & temp_r5) | temp_r4;
+            temp_r1[4] = (temp_r1[4] & temp_r5) | temp_r4;
+            temp_r1[5] = (temp_r1[5] & temp_r5) | temp_r4;
+            temp_r1[6] = (temp_r1[6] & temp_r5) | temp_r4;
+            temp_r1[7] = (temp_r1[7] & temp_r5) | temp_r4;
+        }
+        for (j = 0; j < (arg3 & 7); j++, temp_r1++) {
+            temp_r1[0] = (temp_r1[0] & temp_r5) | temp_r4;
+        }
+        temp_r6 += arg0->unk4;
+    }
+}
+
+void func_08006448(struct struct_08005e54 *arg0, s32 arg1, s32 arg2, s32 arg3, s32 arg4, u16 arg5) {
+    u32 i;
+    u32 j;
+    u16 *temp_r6;
+    u16 *temp_r1;
+    u16 temp_r4 = arg5 << 0xc;
+
+    temp_r6 = &arg0->unk0[arg1] + arg2 * arg0->unk4;
+    for (i = 0; i < arg4; i++) {
+        temp_r1 = temp_r6;
+        for (j = 0; j < (arg3 >> 3); temp_r1 += 8, j++) {
```

Comment:
these for loops kinda look like a fakematch, i feel like theres a simpler way to do it given that j isnt even used here. probably like temp_r1[j][0] etc. with slightly different types

Reply:
Tried to rewrite them, but it doesn't match any other way